src/Datadog.Trace/Configuration/GlobalSettings.cs

```diff
@@ -184,5 +223,21 @@ private static bool TryLoadHostingEnvironmentPath(out string hostingPath)
             hostingPath = default;
             return false;
         }
+
+        private static void LoadVendorFactoryConfigurator()
+        {
+            if (Source.VendorPluginLoaded)
+            {
+                var factoryInterface = typeof(IFactoryConfigurator);
+                var factoryType = Source.VendorPlugin
+                    .GetTypes()
+                    .FirstOrDefault(x => factoryInterface.IsAssignableFrom(x));
+
+                if (factoryType != null)
+                {
+                    Source.FactoryConfigurator = (IFactoryConfigurator)Activator.CreateInstance(factoryType);
```

Comment:
I think that, as a rule of thumb, plugins should EXTEND and not REPLACE the core functionality. Therefore, whenever possible, plugins should be registered in a form of enumerators/decorators/middlewares. Here it is for sure possible. E.g. we may want to register multiple plugins where each adds an additional propagator. E.g. for Datadog, AWS and OpenCensus. More: https://github.com/open-telemetry/opentelemetry-specification/blob/3f07b6669dbbf4625579b5a0f371768f50788cd2/specification/context/api-propagators.md#propagators-distribution

Reply:
Do you mean to scan through every plugin assembly or just forward settings and each plugin assembly will provide what it could?
Any ideas how to override existing logic? For eg some propagators have vendor specific fields not used in core functionality.

Reply:
The plugin e.g. could implement `B3ExtendedPropagator` by overriding the `B3Propagator` and the tracer would use it when `b3extended` is set in `OTEL_PROPAGATORS`.

Reply:
> Do you mean to scan through every plugin assembly or just forward settings and each plugin assembly will provide what it could?

I think that we would have to scan the assemblies and we want to SOMEHOW register stuff from the plugins. Maybe we can even provide the functionality to "Add" or "Replace" via registration API. Take a look at https://blog.ploeh.dk/2014/05/19/di-friendly-framework/ for inspiration